src/pages/home/report/ReportActionContextMenu.js

```diff
@@ -14,31 +18,53 @@ const CONTEXT_ACTIONS = [
     // Copy to clipboard
     {
         text: 'Copy to Clipboard',
-        icon: Clipboard,
+        icon: ClipboardIcon,
+        successText: 'Copied!',
+        successIcon: Checkmark,
+
+        // If return value is true, we switch the `text` and `icon` on
+        // `ReportActionContextMenuItem` with `successText` and `successIcon` which will fallback to
+        // the `text` and `icon`
+        onPress: (action) => {
+            const lastMessage = _.last(lodashGet(action, 'message', null));
+            const html = lodashGet(lastMessage, 'html', '');
+            const text = lodashGet(lastMessage, 'text', '');
+            const isAttachment = text === '[Attachment]';
+            if (!isAttachment) {
+                Clipboard.setString(text);
+            } else {
+                Clipboard.setString(html);
+            }
+            return true;
```

Comment:
How can it return anything else? And then why make this return a boolean at all?

Reply:
my thoughts were like. We don't want to switch to different icons and text for all items. thus each item needs to tell explicitly that does it needs to switch icon. So passing true marks that there is a success.

Reply:
What things need this behavior? I don't see anything like that here as we are always returning `true`.

Reply:
Other press handlers are returning nothing which means don't trigger a `success` state for them. In this function, `setString` is sync which just works. thus I am returning `true` directly.
 https://github.com/Expensify/Expensify.cash/blob/4a4481d12a60dc5dff6ab8b16cf410d8d916444a/src/pages/home/report/ReportActionContextMenu.js#L43-L58.
 
 Do you have any other suggestions to manage `success` state for content items?

Reply:
First, I would suggest implementing this in a PR that needs it so it's more obvious why it is needed. It's hard to suggest an alternative because I am not seeing the context in which this will be used.

Reply:
Yeah. Thanks for clarifying. I was thinking that you would say that. I will remove the `true` check. It's better to be added when other ContextMenuItems' functionality is done.